Split SmartOS related grain off into a seperate module

sjorge opened the pull request:
### What does this PR do?
Moves the SmartOS related grains (both computenode and native zones) into a separate module (salt.grains.smartos)

In addition to this some of the costly execs were reduced from 4 to 2, making those grains a bit faster to be populated. Grain info about the type of vms' is now also included.

For zone an extra grain was also added (```pkgin_repositories```). Originally I wanted to fix the pkgsrcpath grain to list all repositories but that grain is based on an environment value and that one does indeed only hold the main repesitory.

### What issues does this PR fix or reference?
N/A

### Previous Behavior
Grains were in salt.grains.core

### New Behavior
Grains are in salt.grains.smartos, extra grain ```pkgin_repositories``` was added, small performance improvement.

### Tests written?
No

### Commits signed with GPG?
No

sjorge committed:
Remove smartos specific grains from core

This allows the removal of the json module dependancy
NOTE: Not sure how the override of some unavailable smbios grains will
work when they are done in a seperate module. TIAS

sjorge committed:
Moved SmartOS grains to salt.grains.smartos

For the computenode data 3 exec's were condensed into one, execs are
expensive.

sjorge committed:
Also make available info about the type of vms

Commit:
Merge branch 'develop' into smartos_grain_split

cachedout commented on salt/grains/smartos.py:
```diff
@@ -0,0 +1,173 @@
+    return grains
+
+
+def _smartos_zone_data():
```
Can we split this out into separate functions? The complexity of this function is quite high.

sjorge replied:
How about splitting this in 3?
- _smartos_zone_pkgsrc_data
- _smartos_zone_pkgin_data
- _smartos_zone_image_data

?

cachedout replied:
Perfect!

sjorge replied:
@cachedout done

sjorge committed:
Split _smartos_zone_data

This commit splits _smartos_zone_data into 3 functions,
- _smartos_zone_data => zoneid, zonename, imageversion
- _smartos_zone_pkgsrc_data => pkgsrc related data for the zone
- _smartos_zone_pkgin_data => pkgin related data for the zone

sjorge committed:
Merge branch 'smartos_grain_split' of github.com:sjorge/salt into smartos_grain_split

sjorge commented:
Hmm a whole bunch of test failures, all with:
```u'The salt master could not be contacted. Is master running?'```

Can someone give jenkins another run please?

Commit:
Merge branch 'develop' into smartos_grain_split

Commit:
Merge branch 'develop' into smartos_grain_split
